tangos/input_handlers/halo_stat_files/__init__.py

```diff
@@ -206,8 +217,37 @@ def filename(cls, timestep_filename):
         if basename.startswith("snapshot_"):
             timestep_id = int(basename[9:])
             return os.path.join(dirname, "out_%d.list"%timestep_id)
+        elif basename.startswith(("RD","DD")):
```

Comment:
Good point! I've separated all of the DD/RD checks out from the checks for datasets.txt. The new structure is: check for 'snapshot_' format, check for the existence of 'datasets.txt', then check for 'RD/DD' format. I guess we could also move the datasets.txt check above the 'snapshot_' check in case outputs with this format were run with yt's rockstar, but I don't know enough about GADGET outputs to know if that's useful.